src/Extents.jl

```diff
@@ -12,38 +12,58 @@ A wrapper for a `NamedTuple` of tuples holding
 the lower and upper bounds for each dimension of the object.
 
 `keys(extent)` will return the dimension name Symbols,
-in the order the dimensions are used in the object. 
+in the order the dimensions are used in the object.
 
 `values` will return a tuple of tuples: `(lowerbound, upperbound)` for each dimension.
 """
-struct Extent{T<:NamedTuple}
-    bounds::T
+struct Extent{K,V}
+    bounds::NamedTuple{K,V}
+    function Extent{K,V}(bounds::NamedTuple{K,V}) where {K,V}
+        bounds = map(b -> promote(b...), bounds)
+        new{K,typeof(values(bounds))}(bounds)
+    end
 end
 Extent(; kw...) = Extent(values(kw))
+Extent{K}(vals::V) where {K,V} = Extent{K,V}(NamedTuple{K,V}(vals))
+Extent{K1}(vals::NamedTuple{K2,V}) where {K1,K2,V} = Extent(NamedTuple{K1}(vals))
```

Comment:
What's the point of this conversion? Should we check for K1 == K2? Might be good to test these constructors as well.